lib/pure/os.nim

```diff
@@ -1592,10 +1592,17 @@ proc getFilePermissions*(filename: string): set[FilePermission] {.
     else:
       result = {fpUserExec..fpOthersRead}
 
-proc setFilePermissions*(filename: string, permissions: set[FilePermission]) {.
-  rtl, extern: "nos$1", tags: [WriteDirEffect], noWeirdTarget.} =
+proc setFilePermissions*(filename: string, permissions: set[FilePermission],
+                         followSymlinks = true)
+  {.rtl, extern: "nos$1", tags: [ReadDirEffect, WriteDirEffect],
+   noWeirdTarget.} =
   ## Sets the file permissions for `filename`.
   ##
+  ## If ``followSymlinks`` set to true (default) and ``filename`` points to a
+  ## symlink, permissions are set to the file symlink points to.
+  ## ``followSymlinks`` set to false do not affect on Windows and some POSIX
+  ## systems (including Linux) which do not have ``lchmod`` available.
```

Comment:
on ubuntu lchmod is available but compiling a program with it gives a warning: `warning: lchmod is not implemented and will always fail`; maybe this can be reworded:
```nim
## systems (including Linux) on which `lchmod` is either unavailable or always fails,
## given that symlinks there have fixed permisssions.
```